File: fixtures/mut-bytes/tests/bindings/test_mut_bytes.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import uniffi_mut_bytes
import Foundation

// Zero-copy &mut [u8] via UDL [ByMutRef]. Rust writes land in the caller's Data.
var fillMe = Data(count: 4)
fillBytesUdl(buf: &fillMe)
assert(fillMe == Data([0, 1, 2, 3]))

var incMe = Data([1, 2, 3])
incrementBytesUdl(buf: &incMe)
assert(incMe == Data([2, 3, 4]))

// Empty buffer must not crash.
var empty = Data()
fillBytesUdl(buf: &empty)
assert(empty == Data())

File: fixtures/mut-bytes/tests/bindings/test_mut_bytes.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from uniffi_mut_bytes import *

# Zero-copy &mut [u8] via UDL [ByMutRef]. Rust writes land in place.
buf = bytearray(4)
fill_bytes_udl(buf)
assert buf == bytearray([0, 1, 2, 3]), buf

inc = bytearray([1, 2, 3])
increment_bytes_udl(inc)
assert inc == bytearray([2, 3, 4]), inc

# Empty buffer is handled without crashing.
empty = bytearray(0)
fill_bytes_udl(empty)
assert empty == bytearray(0), empty
